### PR TITLE
Improve documentation of `message-bytes-limit`

### PR DESCRIPTION
**Pull request by hendrikmakait**
Addresses
* https://github.com/dask/distributed/pull/7074#discussion_r981070853

**Note**
It just occurred to me that it might be even better to rename `message-bytes-limit` to `gather-bytes-limit` and `transfer_message_bytes_limit` to `transfer_gather_bytes_limit` to highlight their intention, but I don't want to keep changing names all the time.

- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

cc @wence-, @crusaderky

### CODE REVIEW

File: distributed/worker_state_machine.py [hunk 1]
```diff
@@ -1126,10 +1126,11 @@ class WorkerState:
     #: multiple entries in :attr:`~TaskState.who_has` will appear multiple times here.
     data_needed: defaultdict[str, HeapSet[TaskState]]
 
-    #: Number of bytes to fetch from the same worker in a single call to
-    #: :meth:`BaseWorker.gather_dep`. Multiple small tasks that can be fetched from the
-    #: same worker will be clustered in a single instruction as long as their combined
-    #: size doesn't exceed this value.
+    #: Number of bytes to gather from the same worker in a single call to
```
Comment (hendrikmakait) [inline distributed/worker_state_machine.py:1129]: Driveby: Highlight the fine print here as well.

**Review by wence- — APPROVED** [review]
Thanks!

File: distributed/distributed-schema.yaml [hunk 2]
```diff
@@ -349,7 +349,12 @@ properties:
                   - string
                   - integer
                 description: |
-                  The maximum size of a message sent between workers
+                  The maximum amount of data for a worker to request from another in a single ``GatherDep``
+
+                  Tasks are gathered in batches, and if the first task in a batch is larger than this value,
+                  the task will still be gathered to ensure progress. Hence, this limit is not absolute. 
+                  Note that this limit applies to a single ``GatherDep`` and a worker may gather data from 
+                  multiple workers in parallel.
```
Comment (crusaderky) [inline distributed/distributed-schema.yaml:357]: The `GatherDep` class is an implementation detail and shouldn't appear in what is effectively functional documentation. Could you rephrase? e.g. replace \`\`GatherDep\`\`` with "data transfer between workers"?

Comment (hendrikmakait) [reply distributed/distributed-schema.yaml:357]: I've replaced `GatherDep` with _gathering operation_. This maps to the nomenclature we use in our documentation for the worker state machine. A data transfer _may_ be falsely interpreted as a transfer on the comm level, i.e., the transfer of an individual frame, which in turn has its own configuration variable.

**Review by crusaderky — APPROVED** [review]
